ext/har/logger_test.go

```diff
@@ -0,0 +1,187 @@
+package har
+
+import (
+    "io"
+    "net/http"
+    "net/http/httptest"
+    "net/url"
+    "strings"
+    "testing"
+    "time"
+    "sync"
+    "github.com/elazarl/goproxy"
+    "github.com/stretchr/testify/assert"
+    "github.com/stretchr/testify/require"
+)
+
+// ConstantHandler is a simple HTTP handler that returns a constant response
+type ConstantHandler string
+
+func (h ConstantHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+    w.Header().Set("Content-Type", "text/plain")
+    io.WriteString(w, string(h))
+}
+
+// createTestProxy sets up a test proxy with a HAR logger
+func createTestProxy(logger *Logger) *httptest.Server {
+    proxy := goproxy.NewProxyHttpServer()
+    proxy.OnRequest().DoFunc(logger.OnRequest)
+    proxy.OnResponse().DoFunc(logger.OnResponse)
+    return httptest.NewServer(proxy)
+}
+
+// createProxyClient creates an HTTP client that uses the given proxy
+func createProxyClient(proxyURL string) *http.Client {
+    proxyURLParsed, _ := url.Parse(proxyURL)
+    tr := &http.Transport{
+        Proxy: http.ProxyURL(proxyURLParsed),
+    }
+    return &http.Client{Transport: tr}
+}
+
+func TestHarLoggerBasicFunctionality(t *testing.T) {
+    testCases := []struct {
+        name           string
+        method         string
+        body           string
+        contentType    string
+        expectedMethod string
+    }{
+        {
+            name:           "GET Request",
+            method:         http.MethodGet,
+            expectedMethod: http.MethodGet,
+        },
+        {
+            name:           "POST Request",
+            method:         http.MethodPost,
+            body:           `{"test":"data"}`,
+            contentType:    "application/json",
+            expectedMethod: http.MethodPost,
+        },
+    }
+
+    for _, tc := range testCases {
+        t.Run(tc.name, func(t *testing.T) {
+            var wg sync.WaitGroup
+            wg.Add(1)
+            
+            var exportedEntries []Entry
+            exportFunc := func(entries []Entry) {
+                exportedEntries = append(exportedEntries, entries...)
+                wg.Done()
+            }
+            
+            logger := NewLogger(exportFunc, WithExportThreshold(1)) // Export after each request
+            defer logger.Stop()
+
+            background := httptest.NewServer(ConstantHandler("hello world"))
+            defer background.Close()
+
+            proxyServer := createTestProxy(logger)
+            defer proxyServer.Close()
+
+            client := createProxyClient(proxyServer.URL)
+
+            req, err := http.NewRequest(tc.method, background.URL, strings.NewReader(tc.body))
```

Comment:
Use NewRequestWithContext
```suggestion
            req, err := http.NewRequestWithContext(context.Background(), tc.method, background.URL, strings.NewReader(tc.body))
```